lib/rujitsu/date.rb

```diff
@@ -0,0 +1,24 @@
+module RujitsuDate
+
+  def start_of_this_month
+    today = self.today
+    self.new(today.year, today.month)
+  end
+
+  def start_of_last_month
+    today = self.today
+    self.new(today.year, today.month - 1)
```

Comment:
You also have a boundary condition here like you spotted for `start_of_next_month` In January this breaks.

```
> bundle exec irb -rtimecop -rujitsu
irb(main):001:0> Timecop.travel 2000, 1, 1
=> 2000-01-01 00:00:00 +0000
irb(main):002:0> Date.today
=> #<Date: 2000-01-01 ((2451545j,0s,0n),+0s,2299161j)>
irb(main):003:0> Date.start_of_last_month
ArgumentError: invalid date
```